File: node/src/controllers/delete-user-controller.ts
import { Request, Response } from 'express';
import { deleteUserFromDB } from '../models/delete-user-model';

export async function deleteUserController(req:Request, res:Response){
    const { username } = req.query;
    if (!username){
        res.statusCode = 400;
        res.statusMessage = 'Username not provided';
        res.send();
        return;
    }
    try {
        const result = await deleteUserFromDB(username.toString());
        if (result){
            res.statusCode = 200;
            res.statusMessage = 'User deleted';
            res.send();
        } else {
            res.statusCode = 404;
            res.statusMessage = 'User to delete not found';
            res.send();
        }
    } catch (e){
        console.error('🛑 Error processing DELETE User request');
        console.error(e);
        res.statusCode = 500;
        res.send();
    }

}

File: go/controllers/deleteUserController.go
package controllers

import (
	"api-battle/models"
	"fmt"
	"net/http"

	"github.com/gin-gonic/gin"
)

func DeleteUser(c *gin.Context) {
	username, usernameOk := c.GetQuery("username")
	if usernameOk && len(username) > 0 {
		err := models.DeleteUserFromDB(username)
		if err != nil {
			fmt.Println("🛑 Delete user error")
			fmt.Println(err)
			c.Status(http.StatusNotFound)
		} else {
			fmt.Printf("Deleted user %v\n", username)
			c.Status(http.StatusOK)
		}
	} else {
		fmt.Println("🛑 Bad request in delete user, missing username")
		c.Status(http.StatusBadRequest)
	}
}
